onnx/version_converter/convert.cc

```diff
@@ -0,0 +1,122 @@
+#include "onnx/version_converter/convert.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+namespace {
+  static DefaultVersionConverter _version_converter;
+}
+
+ModelProto ConvertVersion(
+    const ModelProto& mp_in,
+    const int target_version) {
+  // Get initial_opsetid from mp_in
+  OpSetID initial_struct(0);
+  for (auto it = mp_in.opset_import().begin(); it != mp_in.opset_import().end(); ++it) {
+    if (it->domain() == "" || it->domain() == "ai.onnx") {
+      initial_struct.setVersion(it->version());
+      break;
+    }
+  }
+  OpSetID target_struct = OpSetID(target_version);
+  return _version_converter.convert_version(mp_in, initial_struct, target_struct);
+}
+
+ModelProto DefaultVersionConverter::convert_version(
+    const ModelProto& mp_in,
+    const OpSetID& initial_version,
+    const OpSetID& target_version) const {
+  const std::string initial_domain = initial_version.domain();
```

Comment:
you might be able to make these two into references if `.domain()` returns a reference. If it returns by value, then you can't because you'd be keeping a reference to a dead object.

Reply:
It does, so I will do that!

Reply:
The sanity check part can be lifted into another function.

Reply:
@Ac2zoom some of the logic can be polished.